src/grid/cubic.py

```diff
@@ -0,0 +1,582 @@
+                )
+            else:
+                raise NotImplementedError(
+                    "Taking mixed derivative while applying the logarithm is not supported."
+                )
+        # Normal interpolation without logarithm.
+        interpolated = x_spline(real_pt[0], real_pt[1], real_pt[2], nu_x)
+        return interpolated
+
+    def coordinates_to_index(self, indices):
+        r"""Convert (i, j, k) integer coordinates to the grid index m.
+
+        Parameters
+        ----------
+        indices : (int, int, int)
+            The ith, jth, kth position of the grid point.
+
+        Returns
+        -------
+        index : int
+            Index of the grid point.
+
+        """
+        n_1d, n_2d = self.shape[2], self.shape[1] * self.shape[2]
+        index = n_2d * indices[0] + n_1d * indices[1] + indices[2]
+        return index
+
+    def index_to_coordinates(self, index):
+        r"""Convert index of grid point to its (i, j, k) coordinates in a cubic grid.
+
+        Cubic grid has a shape of :math:`(N_x, N_y, N_z)` denoting the number of points in
+        :math:`x`, :math:`y`, and :math:`z` directions. So, each grid point has a :math:`(i, j, k)`
+        integer coordinate where :math:`0 <= i <= N_x - 1`, :math:`0 <= j <= N_y - 1`,
+        and :math:`0 <= k <= N_z - 1`.
+
+        Parameters
+        ----------
+        index : int
+            Index of the grid point.
+
+        Returns
+        -------
+        indices : (int, int, int)
+            The corresponding :math:`(i, j, k)` integer coordinates in a cubic grid.
+
+        """
+        if not index >= 0:
+            raise ValueError(
+                f"Argument index should be a positive integer, got {index}"
+            )
+        n_1d, n_2d = self.shape[2], self.shape[1] * self.shape[2]
+        i = index // n_2d
+        j = (index - n_2d * i) // n_1d
+        k = index - n_2d * i - n_1d * j
+        return i, j, k
+
+
+class Tensor1DGrids(_CubicGrid):
+    r"""Tensor1DGrids : Tensor product of three one-dimensional grids."""
+
+    def __init__(self, oned_x, oned_y, oned_z):
+        r"""Construct Tensor1DGrids by tensor product of three one-dimensional grids.
+
+        Parameters
+        ----------
+        """
+        if not isinstance(oned_x, OneDGrid):
+            raise TypeError(
+                f"Argument oned_x should be an instance of `OneDGrid`, got {type(oned_x)}"
+            )
+        if not isinstance(oned_y, OneDGrid):
+            raise TypeError(
+                f"Argument oned_y should be an instance of `OneDGrid`, got {type(oned_y)}"
+            )
+        if not isinstance(oned_z, OneDGrid):
+            raise TypeError(
+                f"Argument oned_z should be an instance of `OneDGrid`, got {type(oned_z)}"
+            )
+
+        # number of points in x, y, and z direction of the cubic grid
+        shape = (oned_x.size, oned_y.size, oned_z.size)
+
+        # Construct 3D set of points
+        points = (
```

Comment:
@Ali-Tehrani Can you please add a comment here to clarify in which order the points are stored? Is it (z, y, x)?